docs/source/extending/contents.rst

```diff
@@ -200,6 +200,25 @@ You may be required to specify a Checkpoints object, as the default one,
 ``FileCheckpoints``, could be incompatible with your custom 
 ContentsManager.
 
+
+Chunked Saving
+~~~~~~~~~~~~~~
+
+The contents API allows for "chunked" saving of files, i.e.
+saving/transmitting in partial pieces:
+
+* This can only be used when the ``type`` of the model is ``file``.
+* The model should be as otherwise expected for
+  :meth:`~manager.ContentsManager.save`, with an added field ``chunk``.
+* The value of ``chunk`` should be an integer starting at ``1``, and incrementing
+  for each subsequent chunk, except for the final chunk, which should be
+  indicated with a value of ``-1``.
+* The model returned from using :meth:`~manager.ContentsManager.save` with
+  ``chunk`` should be treated as unreliable for all chunks except the final one.
+* Any interaction with a file being saved in a chunked manner is unreliable
+  until the final chunk has been saved. This includes directory listings.
```

Comment:
Thanks Vidar.

Looking at this area of the code, I see the pre and post save hooks are invoked on the first and last chunks.  While that should be obvious, I wonder if it's worth an explicit statement that the hook invocations are relative to the overall file and not per chunk/request.

Reply:
As these hooks are not mentioned in the current docs at all, I'm not entirely sure what to write, or what context to add it in (it would be weird to only mention them in passing when discussing something else). Any suggestions for what to write are welcome (either as a comment or as a commit on the PR branch).

Reply:
I added an attempt below for now.

Reply:
Ah - I didn't realize the hooks weren't documented - so I apologize for bringing this up at all.